book/tut2_spec_pd_gpd.ipynb

```diff
@@ -0,0 +1,504 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cc67a58c-45f3-4819-9df4-04e5aa9d06f7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gpd_rivers_org.crs\n",
+    "gpd_rivers = gpd_rivers_org.to_crs('EPSG:4326')\n",
+    "gpd_rivers.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c6ed1e0c-3b1c-485a-9640-9e8098e573f1",
+   "metadata": {},
+   "source": [
+    "#### 2.1.2 Create a geographical map of the rivers\n",
+    "\n",
+    "Let's plot the rivers represented on top of a map. The `geopandas.DataFrame` can be directly passed to the `data` parameter of [`pygmt.Figure.plot`](https://www.pygmt.org/v0.13.0/api/generated/pygmt.Figure.plot.html). Use the `pen` parameter to adjust the lines. The string argument has the form \"*width*,*color*,*style*\". Different line styles are explained in the Gallery example [**Line styles**](https://www.pygmt.org/v0.13.0/gallery/lines/linestyles.html)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "250d19db-daaf-459d-b8b6-fba5752eec62",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig = pygmt.Figure()\n",
+    "\n",
+    "fig.coast(\n",
+    "    projection=\"M10c\", \n",
+    "    region=[-10, 30, 35, 57],\n",
+    "    land=\"gray99\",\n",
+    "    shorelines=\"1/0.1p,gray50\",\n",
+    "    frame=True,\n",
+    ")\n",
+    "\n",
+    "fig.plot(data=gpd_rivers, pen=\"0.5p,steelblue,solid\")\n",
+    "\n",
+    "fig.show(dpi=img_dpi)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "94058ab7-657d-49b4-a8f5-2c1b8533a4a3",
+   "metadata": {},
+   "source": [
+    "#### 2.1.3 Plot subsets of the rivers differently\n",
+    "\n",
+    "Now we want to filter the dataset based on the river length and plot the subsets differently."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9de1b7a9-0e87-4368-8df1-7064e7ac616c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig = pygmt.Figure()\n",
+    "\n",
+    "fig.coast(\n",
+    "    projection=\"M10c\", \n",
+    "    region=[-10, 35, 35, 58],\n",
+    "    land=\"gray99\",\n",
+    "    shorelines=\"1/0.1p,gray50\",\n",
+    "    frame=True,\n",
+    ")\n",
+    "\n",
+    "# Split the dataset into two subsets of shorter and longer rivers\n",
+    "# Feel free to play around with the limit\n",
+    "len_limit = 700000  # in meters\n",
+    "gpd_rivers_short = gpd_rivers[gpd_rivers[\"Shape_Leng\"] < len_limit]\n",
+    "gpd_rivers_long = gpd_rivers[gpd_rivers[\"Shape_Leng\"] > len_limit]\n",
+    "\n",
+    "# Add a heading (+H) to the legend and adjust the font size (+f) to 9 points\n",
+    "fig.plot(data=gpd_rivers_short, pen=\"0.5p,orange\", label=f\"shorter {len_limit} m+Hriver length+f9p\")\n",
+    "fig.plot(data=gpd_rivers_long, pen=\"0.5p,darkred\", label=f\"longer {len_limit} m\")\n",
+    "\n",
+    "fig.legend(position=\"jLT+o0.1c\", box=\"+gwhite@30+p0.1p,gray50\")\n",
```

Comment:
Since `fig.legend` isn't explained in the first tutorial, could you add some text to explain it before the code cell? You can just link to https://www.pygmt.org/v0.13.0/api/generated/pygmt.Figure.legend.html and/or one of the PyGMT gallery examples using legend.

Reply:
Makes sense! I have added some documentation in commit https://github.com/GenericMappingTools/agu24workshop/pull/4/commits/909f713ffa838452b364289b49180d4db8fc09b7.